RDKShell/RDKShell.cpp

```diff
@@ -544,12 +550,68 @@ namespace WPEFramework {
             }
             return exist;
         }
+        static bool isRebootClientExists(std::string client)
+        {
+            bool exist = false;
+			std::map<std::string, std::string>::iterator it;
+			for(it=reset_reason.begin();it!=reset_reason.end();it++)
+			{
+			  if(it->first == client)
+			  {
+				 exist=true;
+				 break;
+			  }
+			}
+			return exist;
+		}
 
         void RDKShell::MonitorClients::StateChange(PluginHost::IShell* service)
         {
             if (service)
             {
                 PluginHost::IShell::state currentState(service->State());
+                PluginHost::IShell::reason currentreason(service->Reason());
+                 
```

Comment:
move below block to separate function populateRebootReason and pass needed arguments